### PR TITLE
Support object ref args

### PR DESCRIPTION
**Pull request by qicosmos**
## Why are these changes needed?

Support ObjectRef arguments of Ray Task, this PR only support local mode, next PR will support cluster mode.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests

### CODE REVIEW

**Review by mwtian — COMMENTED** [review]
Looks ok in general. Will defer to others for API.

**Review by SongGuyang — APPROVED** [review]
Please ensure address the left comments before merge.

**Review by raulchen — APPROVED** [review]

**Pull request merged**
